arithmetization/src/main/java/net/consensys/linea/zktracer/module/oob/OobOperation.java

```diff
@@ -487,6 +464,28 @@ public void populateColumnsForPrecompile(MessageFrame frame) {
     }
   }
 
+  // Support method for MODEXP
+  public static int computeExponentLog(Bytes paddedCallData, int cds, int bbs, int ebs) {
+    Preconditions.checkArgument(paddedCallData.size() >= 96);
+
+    // pad paddedCallData to 96 + bbs + ebs
+    final Bytes doublePaddedCallData =
+        cds < 96 + bbs + ebs ? rightPadTo(paddedCallData, 96 + bbs + ebs) : paddedCallData;
+
+    final BigInteger leadingBytesOfExponent =
+        doublePaddedCallData.slice(96 + bbs, min(ebs, 32)).toUnsignedBigInteger();
+
+    if (ebs <= 32 && leadingBytesOfExponent.signum() == 0) {
+      return 0;
+    } else if (ebs <= 32 && leadingBytesOfExponent.signum() != 0) {
+      return log2(leadingBytesOfExponent, RoundingMode.FLOOR);
+    } else if (ebs > 32 && leadingBytesOfExponent.signum() != 0) {
+      return 8 * (ebs - 32) + log2(leadingBytesOfExponent, RoundingMode.FLOOR);
+    } else {
+      return 8 * (ebs - 32);
```

Comment:
Re-reading this makes me wonder whether we have tests where we deliberately set the `leadingBytesOfExponent` to zero ...